src/benchttp/metrics.ts

```diff
@@ -0,0 +1,18 @@
+import { GoDuration } from './common'
+
+export type MetricField = DurationField | NumberField
+
+export type DurationField = SingleMetricField<DurationMetricId, GoDuration>
+
+export type NumberField = SingleMetricField<NumberMetricId, number>
+
+interface SingleMetricField<M extends MetricId, T> {
+  id: M
+  value: T
+}
+
+type MetricId = MetricField['id']
```

Comment:
`MetricId` circularly references itself through MetricId` -> `MetricField` -> `DurationField | NumberField` -> `SingleMetricField<MetricId>`.

I suggest to simply define `MetricId` as the union of all ids.

```suggestion
type MetricId = DurationMetricId | NumberMetricId
```

Reply:
I noticed that, but TypeScript seems to handle that pretty well.
Let's change it back to the union, I agree it can be confusing.

Reply:
Maintenance would be small as an error would pop quickly if they get out of sync.
My main point against it is the circumvoluted way of defining the types.

I let you choose what you prefer ! :)

Reply:
I'll let @alex-mngt decide then 😛

Reply:
Finally decided to move on with the current `MetricField['id']`, we'll have the occasion to discuss it again because these definitions will probably change in a near future (see https://github.com/benchttp/engine/pull/41).
Merging anyway, so we can benefit from this types without being blocked by the linked PR.